src/main/scala/translating/SemanticsLoader.scala

```diff
@@ -0,0 +1,470 @@
+package translating
+import Parsers.SemanticsParser.*
+import com.google.protobuf.ByteString
+import Parsers.*
+
+import java.util.Base64
+import scala.jdk.CollectionConverters.*
+import ir.*
+
+import scala.collection.mutable
+import scala.collection.mutable.Map
+import scala.collection.mutable.ArrayBuffer
+import com.grammatech.gtirb.proto.Module.ByteOrder.LittleEndian
+
+import scala.collection.immutable
+
+class SemanticsLoader(targetuuid: ByteString, parserMap: immutable.Map[String, Array[Array[StmtContext]]], blkCount: Int) extends SemanticsBaseVisitor[Any] {
+
+  private var cseMap = mutable.Map[String, IRType]()
+  private var varMap = mutable.Map[String, IRType]()
+  private var instructionCount = 0
+
+  def chrisifyUUID(uuid: ByteString): String = { // This probably needs a better name, but :/
+    Base64.getEncoder.encodeToString(uuid.toByteArray)
+  }
+
+  def createStatements(): ArrayBuffer[Statement] = {
+    visitInstructions(parserMap(chrisifyUUID(targetuuid)))
+  }
+
+  def visitInstructions(stmts: Array[Array[StmtContext]]): ArrayBuffer[Statement] = {
+
+    val statements: Array[Statement] = stmts.flatMap { elem => 
+      cseMap = cseMap.empty
+      varMap = varMap.empty
+      val stmts = elem.flatMap {
+        s => s match {
+          case a if a.assignment_stmt() != null =>
+            visitAssignment_stmt(a.assignment_stmt())
+          case c if c.call_stmt() != null =>
+            Option(visitCall_stmt(c.call_stmt()))
+          case c if c.conditional_stmt() != null =>
+            Option(visitConditional_stmt(c.conditional_stmt()))
+          case _ => ???
+        }
+      }
+      instructionCount += 1
+      stmts
+    }
+
+    statements.to(ArrayBuffer)
+  }
+
+  def visitAssignment_stmt(ctx: Assignment_stmtContext): Option[Statement] = {
+    ctx match
+      case a: AssignContext => Option(visitAssign(a))
+      case c: ConstDeclContext => Option(visitConstDecl(c))
+      case v: VarDeclContext => Option(visitVarDecl(v))
+      case v: VarDeclsNoInitContext => visitVarDeclsNoInit(v)
+      case a: AssertContext => Option(visitAssert(a))
+  }
+
+  override def visitAssert(ctx: AssertContext): Assert = {
+    val expr = visitExpr(ctx.expr)
+    Assert(expr)
+  }
+
+  override def visitCall_stmt(ctx: Call_stmtContext): MemoryAssign = {
+    val mem = Memory("mem", 64, 8) // yanked from BAP
+    val addr = visitExpr(ctx.expr(1))
+    val value = visitExpr(ctx.expr(4))
+    val size = visitExpr(ctx.expr(2)).asInstanceOf[IntLiteral].value.toInt * 8
+    val memstore = MemoryStore(mem, addr, value, Endian.LittleEndian, size)
+    MemoryAssign(mem, memstore)
+  }
```

Comment:
This doesn't seem like a robust way of handling `Stmt_TCall`s. From what I can gather, memory stores are represented as calls to the procedure `Mem.set.0`, so we should at least check that a call is to that rather than just assuming. Do we have any sort of definition of `Mem.set.0` anywhere?

Reply:
This is true. I'll add in a check for that. We don't really have a defintion for mem.set.0, apparently it's just some structure that Aslp generates when it reaches something it doesn't know how to do, like memory stores or atomics. 

The Bap-Asli plugin mentions two more functions, "AtomicStart: and "AtomicEnd" but i have no idea how those are currently translated/ if they even are.